### ISSUE TITLE
After config.tsx is changed config.js is not re-generated

### ISSUE THREAD

**Issue by hboon** [onivim/oni#1934]
After working for a while, changes to `config.tsx` no longer regenerates `config.js`. Tried restarting, stripping down `config.tsx`, touching `config.js`, etc. It started working again after I chose `Preferences` > `Edit Oni config` (which also help verify that I was editing the right copy of `config.tsx`.

Can't reproduce anymore.

v0.3.1

**Comment by hboon** [comment]
Can reproduce again on master (v0.3.2 / 26d015cd074fdc4f02e597725bfef9e433ad3c70).

This time `Preferences` > `Edit Oni config` doesn't fix it.

**Comment by badosu** [comment]
@hboon Can you still reproduce this error on v0.3.4?

**Comment by hboon** [comment]
@badosu Yes, I can reproduce it in v0.3.4, download from the Releases page.

**Comment by badosu** [comment]
@hboon Are you sure there are no compilation issues in your `config.tsx` file?

If backup your current config and start a new one, does the new one not change the configuration?

**Comment by hboon** [comment]
@badosu Here's what I just tried:

1. Open config.tsx and config.js in One itself.
2. Change config.tsx to:

```
import * as React from "/Applications/Oni.app/Contents/Resources/app/node_modules/react"
import * as Oni from "/Applications/Oni.app/Contents/Resources/app/node_modules/oni-api"

export const activate = (oni: Oni.Plugin.Api) => {
    console.log("config activated57")
}
```
3. Switch to the config.js tab (it's not refreshed. I have `vim.setting.autoread` enabled)
4. Run: `bd`
5. open config.js again and it's still not refreshed

I repeated the above after starting a new Oni window (since most of the config has been removed, this is pretty much a blank slate?) with the same result.

Maybe I missed something obvious?

**Comment by badosu** [comment]
Are you editing `~/.config/oni/config.tsx`?

**Comment by hboon** [comment]
Yes

**Comment by Akin909** [comment]
I find sometimes that to get the transpilation to work I have to got via the `QuickOpen` menu to `Edit User Config` and then make the change in the split and save it.

Indicative of a bug no doubt, but maybe an interim solution?

**Comment by hboon** [comment]
I just tried that (`QuickOpen`) and `Preferences` > `Edit Oni Config` and while the file is opened, editing it doesn't update `config.js`.

It's no biggie for now though, I'm not making much changes to my configuration for now. So if no one else is observing this issue, maybe it's just me.

**Comment by parkerault** [comment]
@hboon can you open Help > Developer Tools before you try to save your config.tsx file and see if it generates any error messages when you save?

It should output the following if successful:
<img width="1404" alt="image" src="https://user-images.githubusercontent.com/78856/40738226-98173bfa-6408-11e8-8d97-a0b520f07c90.png">

**Comment by hboon** [comment]
When I move to a tab which has `config.tsx` open, I see this in the console (set to verbose):

<img width="543" alt="screen shot 2018-05-31 at 10 41 41 am" src="https://user-images.githubusercontent.com/56189/40758301-3aed975a-64bf-11e8-9e87-8a4792a12f8d.png">

I see something similar (if not exactly the same) when I move to tab with `config.js` open. When I save `config.tsx`, nothing else gets logged to the console.

When I save `config.js` instead, `console.log()` calls in there get called correctly.

I see different messages if I go to a `.swift` file or `.h`. Something about "No language server registered". So it seems to be working otherwise.

**Comment by elentok** [comment]
I can reproduce on 0.3.4 from releases and on master as well

**Comment by iamkarlson** [comment]
So, is there any workaround till fix is in place?

**Comment by Towerism** [comment]
I figured out the problem at least for me: My ~/.config/oni directory is symlinked to my dotfiles repo. If I remove the symbolic link and copy over my config.tsx it will compile just fine.

**Comment by hboon** [comment]
FWIW, My `~/.config/` is symlink-ed a folder in my Dropbox.

**Comment by Towerism** [comment]
@hboon My workaround probably won't work for you. The issue still stands that oni fails to compile config.tsx properly when the `~/.config/oni` directory (and therefore any of its parent directories) is symlinked.

**Comment by hboon** [comment]
@Towerism that's good though, a least if it's reproducible, someone can fix it :)

**Comment by Akin909** [comment]
@hboon and @Towerism one option as a workaround is to set an ENV variable in your `.bashrc` or `.zshrc` which is `$ONI_CONFIG_FILE` if the variable is present oni will use that as the path to your config, let me know if that helps at all, will look into getting it to follow symlinks when I get a chance

**Comment by Towerism** [comment]
@Akin909 This worked for me: `export ONI_CONFIG_FILE=~/.dotfiles/oni/.config/oni/config.tsx`.

@hboon I restarted my X11 session to make sure that the environment variable would be picked up by oni.
